### ISSUE TITLE
clearStore erase pictures from FS but entries property is not updated properly

### ISSUE THREAD

**Issue by robinbiondi** [code-and-co/react-native-image-offline#12]
Great Work
-----------
Hello,
I am so happy to find your package because it suits perfectly my needs. Thank you for this great work!

Problem
--------

Anyway, I was trying to clear the store before re-downloading  images but when doing it, I got an error saying that the images I re-download already exists and then when I try to display the image, I got an error: `Could not find image: file://blablabla....`

Observation
------------
Looking through the code, I noticed that here: https://github.com/code-and-co/react-native-image-offline/blob/09f318773d5eb3ac1bf8e4ddccc41f25ba23c036/src/OfflineImageStore.js#L89
`this. entries = {}` isn't doing anything (If i log `this.entries` on the line after, it is not empty)

Then, I noticed that the `instance` was frozen before being export: https://github.com/code-and-co/react-native-image-offline/blob/09f318773d5eb3ac1bf8e4ddccc41f25ba23c036/src/OfflineImageStore.js#L368

As a result, it is not possible to reassign property like `this.entries = {}`

When I comment this line, it works fine.

Solution
--------
I identified two solutions,

The first and easiest solution would be to remove the line L368 (the freeze of the instance) but I am not sure why it is done like that and I'm not aware of side effects

The second solution would be to rework clearStore function. Iterating on `this.entries` and deleting each keys one by one:

```javascript
delete this.entries[blabla]
```

Conclusion
-----------
I would be happy to develop the fix because I'd like to use it quickly in my project.
Let me know which solution you like the most.
Waiting for your answer, I will be implementing the second one.

Have a nice day!

Robin

**Comment by Jayasagar** [comment]
Hi Robin,

I am so happy that you liked it and useful for you! I really like the issue description and structure 👍 Thanks for it.

I am sorry, not sure whether we tested well with `clearStore` functionality. 

However, I would recommend using the second solution, i.e. delete each entry :) 

I used `freeze` to make the `OfflineImageStore` instance as a Singleton object. So that everytime you import and use the `OfflineImageStore` instance, it should always same one so we will application wide state.
